llama-index-integrations/vector_stores/llama-index-vector-stores-lancedb/llama_index/vector_stores/lancedb/base.py

```diff
@@ -213,26 +301,45 @@ def query(
                     "Use kwargs only for lancedb specific items that are "
                     "not supported via the generic query interface."
                 )
-            where = _to_lance_filter(query.filters)
+            where = _to_lance_filter(query.filters, self._metadata_keys)
         else:
             where = kwargs.pop("where", None)
 
-        table = self._connection.open_table(self.table_name)
+        query_type = kwargs.pop("query_type", "vector")
+
+        if query_type == "vector":
+            _query = query.query_embedding
+        else:
+            self._table.create_fts_index(self.text_key, replace=True)
```

Comment:
This will create index every time query is called, let's keep track of index if that's created in an instance variable and not recreate index every time

Reply:
so this is also done because if the user adds more data, a full reindex is needed from our(lance) end, creating it every time is also not the best way but we would have to handle this in the future.
Resolving it as per your comment as of now thanks.

Reply:
yes I actually just did that haha